tests/test_configuration.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import sys
+import tempfile
+from pathlib import Path
+from unittest.mock import patch
+
+import platformdirs
+
+from variantlib.configuration import ConfigEnvironments
+from variantlib.configuration import VariantConfiguration
+from variantlib.configuration import get_configuration_files
+from variantlib.constants import CONFIG_FILENAME
+from variantlib.models.configuration import VariantConfiguration as ConfigurationModel
+from variantlib.models.variant import VariantFeature
+from variantlib.models.variant import VariantProperty
+
+
+def dict_to_toml(d: dict, indent: int = 0) -> str:
+    lines = []
+    for key, value in d.items():
+        if isinstance(value, dict):
+            lines.append(f"\n{' ' * indent}[{key}]")
+            lines.append(dict_to_toml(value, indent))
+        elif isinstance(value, list):
+            items = ", ".join(f'"{v}"' if isinstance(v, str) else str(v) for v in value)
+            lines.append(f"{' ' * indent}{key} = [{items}]")
+        elif isinstance(value, str):
+            lines.append(f'{" " * indent}{key} = "{value}"')
+        else:
+            raise TypeError(f"Unsupported type for key '{key}': {type(value)}")
+    return "\n".join(lines)
+
+
+def test_reset():
+    VariantConfiguration._config = ConfigurationModel.default()  # noqa: SLF001
```

Comment:
Perhaps we should disable `SLF001` entirely for tests?

Reply:
Not sure how to do that - And it's not in variantlib anyway - Feel free to contribute that to the monorepo